staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -367,8 +368,9 @@ func (p *smpPatcher) applyPatchToCurrentObject(currentObject runtime.Object) (ru
 	}
 
 	if p.fieldManager != nil {
-		newObj, err = p.fieldManager.Update(currentObject, newObj, "smPatcher")
-		if err != nil {
+		if obj, err := p.fieldManager.Update(currentObject, newObj, "smPatcher"); err == nil {
+			newObj = obj
+		} else {
 			klog.Warningf("FieldManager: Failed to update object managed fields: %v", err)
```

Comment:
surprised this isn't an error

Reply:
Fixed.

Reply:
We wanted to make sure that the feature couldn't break any other verbs besides apply so we are just logging the errors rather than returning them

Reply:
I actually meant I was surprised it didn't return an error. Jenny's comment made sense, might be good to include that reasoning in a comment in the code in a future change (just logging an error there looks like a mistake)